application/src/main/java/org/togetherjava/tjbot/commands/filesharing/DeleteGistCommand.java

```diff
@@ -0,0 +1,57 @@
+package org.togetherjava.tjbot.commands.filesharing;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public final class DeleteGistCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(DeleteGistCommand.class);
+
+    private static final String MESSAGE_ID_OPTION = "message-id";
+
+    public DeleteGistCommand(@NotNull Config config) {
+        super("remove-gist", "Deletes gist's from auto file-sharing feature!",
+                SlashCommandVisibility.GUILD);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandInteractionEvent event) {
+        String messageId = event.getOption(MESSAGE_ID_OPTION).getAsString();
+
+        Message gistMessage = event.getMessageChannel().retrieveMessageById(messageId).complete();
```

Comment:
what if used in a different channel? if this is a limitation you cant get around, then it has to be described to the user in the command description or they will be super confused

Reply:
you stepped into the next multithreading pitfall. `complete()` blocks the current (JDA) thread. an absolute no-go.

instead, use the returned RestAction and its chaining operations `map`. RestAction is JDAs CompletableFuture.

From there on, all your next actions will be in this chain instead.

Also, you need to react on the failure of the message retrieval to provide the user with good UX in case they either gave the wrong ID or used the command in the wrong channel. They need a proper error message that explains them what they did wrong.